doc/Reactive/Reactive.01-concept.md

```diff
@@ -0,0 +1,68 @@
+# Concept
+
+When asynchronously loading a data, the standard pattern is to use a `Task<T>`. A _task_ represents data which  will be available in the future:
+```csharp
+public async Task<decimal> GetShippingCost(CancellationToken ct)
+{
+	var country = SelectedCountry;
+	var cost = await _shippingService.GetShippingCost(country);
+
+	return cost;
+}
+```
+An issue here is that `Task<T>` represents only one value, data must manually be fetched again each time one of its dependencies is updated. For instance, here, each time the user updates the selected country, `GetShippingCost` has to be manually re-invoked and the UI updated.
+
+A solution to this would be to use `IObservable<T>` or `IAsyncEnumerable<T>`. Both are representing a stream of value. The example above can be written like this using `IObservable<T>`:
+```csharp
+public IObservable<Country> SelectedCountry { get; }
+
+public IObservable<decimal> ShippingCost => _selectedCountry.SelectAsync(country => _shippingService.GetShippingCost(country));
+```
+Or with `IAsyncEnumerable`:
+```csharp
+public async IAsyncEnumerable<decimal> GetShippingCost([EnumerationCancellation] CancellationToken ct = default)
+{
+	await foreach (var country in SelectedCountry)
+	{
+		yield return await _shippingService.GetShippingCost(country);
+	}
+}
+```
+
+But in both cases, if there is any exception the stream will be broken. This means that for instance in example above, if it is not possible to compute the shipping cost for a given country for any reason (network issue, invalid country, …) the stream of data will be terminated, and selecting another country won’t have any effect.
+
+Also, when a dependency is being updated and we may need to do some asynchronously work, like update a projection. In our example, we asynchronously get the updated shipping cost when country is changed. From a UI perspective, it would be great to have a visual indication that the shipping cost is being re-computed for the newly chosen country.
```

Comment:
```suggestion
Also, when a dependency is being updated and we may need to do some asynchronous work, like update a projection. In our example, we asynchronously get the updated shipping cost when the country is changed. From a UI perspective, it would be great to have a visual indication that the shipping cost is being re-computed for the newly chosen country.
```